### PR TITLE
Waymo dataset integration

### PR DESCRIPTION
**Pull request by st235**
### Prerequisites

The original Waymo dataset should be converted to the enhanced format with more meta-information, like, **frame** or **scene** ids.

[Waymo Preprocessing Tools](https://github.com/st235/waymo_preprocessing) were used to enhance the dataset. To run the tool use:

```python
python waymo_processor.py [path to waymo dataset root] --save_dir [path to the output directory]
```

### Patched objects

| Object 1 | Object 2 | Object 3 |
| ------ | ------ | ------ |
| Screenshot 2023-09-27 003333 | Screenshot 2023-09-27 003428 | Screenshot 2023-09-27 003447 |

### Patched scenes

| Original scene | Patched scene |
| ------ | ------ |
| Screenshot 2023-09-27 012058 | Screenshot 2023-09-27 021017 |  

### Run

```python
python patch_scene.py --dataset waymo --dataroot [path to processed waymo dataset]
```

### CODE REVIEW

**Comment by st235** [comment]
@mariarzv ptal

File: main.py [hunk 1]
```diff
@@ -4,14 +4,26 @@
...
+
+
 def main():
-    dataset: Dataset = NuscenesDataset(version='v1.0-mini', dataroot='./temp/nuscenes')
+    dataset = __create_dataset('waymo')
 
-    scene_id = '0'
+    print('Scenes:', dataset.scenes)
+
+    scene_id = 'training_segment-10023947602400723454_1120_000_1140_000_with_camera_labels'
```
Comment (mariarzv) [inline main.py:26]: is that waymo specific? but it's in main?

File: src/utils/geometry_utils.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    # Translate
+    x, y, z = centers_xyz
+    corners[0, :] = corners[0, :] + x
+    corners[1, :] = corners[1, :] + y
+    corners[2, :] = corners[2, :] + z
+
+    return corners
+
+
+def points_in_box(center_xyz: np.ndarray,
```
Comment (mariarzv) [inline src/utils/geometry_utils.py:58]: i have the same method for once, but it's a bit different, maybe this can be named points_in_box_waymo?

**Review by mariarzv — COMMENTED** [review]
very minor comments, i don't insist on making any changes though

File: main.py (hunk 1 above)
Comment (st235) [reply main.py:26]: `main` is just a playground for us to test: it works faster than `patch_scene` and has hard-coded values for ids. So should be fine to change values in here.

File: src/utils/geometry_utils.py (hunk 2 above)
Comment (st235) [reply src/utils/geometry_utils.py:58]: this method is unrelated to **waymo**: it checks whether the points are in the box specified by center and dimensions. Can  you use this method? I believe we should try to minimize the amount of geometry utils.

File: main.py (hunk 1 above)
Comment (mariarzv) [reply main.py:26]: ok!

File: src/utils/geometry_utils.py (hunk 2 above)
Comment (mariarzv) [reply src/utils/geometry_utils.py:58]: it's very similar, could reuse it

**Review by mariarzv — APPROVED** [review]

**Pull request merged**